File: arrays/00-2DArray.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the hourglassSum function below.
def hourglassSum(arr):
    max = -sys.maxsize;
    for i in range(1, 5):
        for j in range(1, 5):
            hgsum = sum(arr[i-1][j-1:j+2]) + arr[i][j] + sum(arr[i+1][j-1:j+2])
            if hgsum > max:
                max = hgsum
            
    return max
        

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))

    result = hourglassSum(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: arrays/00-2DArray.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function hourglassSum(arr) {
  var maxSum = -999999;
  for (var i = 1; i < arr.length - 1; i++) {
    for (var j = 1; j < arr[0].length - 1; j++) {
      var currentSumTop = 0;
      var currentSumMiddle = arr[i][j];
      var currentSumBottom = 0;
      
      for (var k = j - 1; k <= j + 1; k++) {
        currentSumTop += arr[i-1][k]; 
      }

      for (var k = j - 1; k <= j + 1; k++) {
        currentSumBottom += arr[i+1][k]; 
      }

      maxSum = Math.max(maxSum, currentSumTop + currentSumMiddle + currentSumBottom);
    }
  }

  return maxSum;
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    let arr = Array(6);

    for (let i = 0; i < 6; i++) {
        arr[i] = readLine().split(' ').map(arrTemp => parseInt(arrTemp, 10));
    }

    let result = hourglassSum(arr);

    ws.write(result + "\n");

    ws.end();
}
